bin/templatemap_generator.php

```diff
@@ -0,0 +1,111 @@
+    require __DIR__ . '/../vendor/autoload.php';
+} elseif (file_exists(getcwd() . '/vendor/autoload.php')) {
+    // Root project is current working directory
+    require getcwd() . '/vendor/autoload.php';
+} elseif (file_exists(__DIR__ . '/../../../autoload.php')) {
+    // Relative to composer install
+    require __DIR__ . '/../../../autoload.php';
+} else {
+    fwrite(STDERR, "Unable to setup autoloading; aborting\n");
+    exit(2);
+}
+
+$help = <<< EOH
+Generate template maps.
+
+Usage:
+
+templatemap_generator.php [-h|--help] templatepath files...
+
+--help|-h                    Print this usage message.
+templatepath                 Path to templates relative to current working
+                             path; used to identify what to strip from
+                             template names.
+files...                     List of files to include in the template
+                             map, relative to the current working path.
+
+The script assumes that paths included in the template map are relative
+to the current working directory.
+
+The script will output a PHP script that will return the template map
+on successful completion. You may save this to a file using standard
+piping operators; use ">" to write to/ovewrite a file, ">>" to append
+to a file (which may have unexpected and/or intended results; you will
+need to edit the file after generation to ensure it contains valid
+PHP).
+
+We recommend you then include the generated file within your module
+configuration:
+
+  'template_map' => include __DIR__ . '/template_map.config.php',
+
+Examples:
+
+  # Create a template_map.config.php file in the Application module's
+  # config directory, relative to the view directory, and only containing
+  # .phtml files; overwrite any existing files:
+  $ cd module/Application/config/
+  $ ../../../vendor/bin/templatemap_generator.php ../view/**/*.phtml > template_map.config.php
+EOH;
+
+// Called without arguments
+if ($argc < 2) {
+    echo $help;
+    exit(2);
+}
+
+// Requested help
+if (in_array($argv[1], ['-h', '--help'], true)) {
+    echo $help, "\n";
+    exit(0);
+}
+
+// Not enough arguments
+if ($argc < 3) {
+    echo $help;
+    exit(2);
+}
+
+$basePath = $argv[1];
+$files    = array_slice($argv, 2);
+$map      = [];
+$realPath = realpath($basePath);
+
+$entries = array_map(function ($file) use ($basePath, $realPath) {
+    $file     = str_replace('\\', '/', $file);
+
+    $template = (0 === strpos($file, $realPath))
+        ? substr($file, strlen($realPath))
+        : $file;
+
+    $template = (0 === strpos($template, $basePath))
+        ? substr($template, strlen($basePath))
+        : $template;
+
+    $template = preg_match('#(?P<template>.*?)\.[a-z0-9]+$#i', $template, $matches)
+        ? $matches['template']
+        : $template;
+
+    $template = str_replace('\\', '/', $template);
+    $template = preg_replace('#^\.*/#', '', $template);
+    $file     = sprintf('__DIR__ . \'%s\'', $file);
```

Comment:
I'm not seeing the same behavior as you. I tried just now using the same arguments (though with a path to my local working checkout of zend-view for the script), and received the following:

```php
<?php
return [
    'application/index/index' => __DIR__ . '../view/application/index/index.phtml',
    'error/404' => __DIR__ . '../view/error/404.phtml',
    'error/index' => __DIR__ . '../view/error/index.phtml',
    'layout/layout' => __DIR__ . '../view/layout/layout.phtml',
    'partial/breadcrumb' => __DIR__ . '../view/partial/breadcrumb.phtml',
    'partial/paginator' => __DIR__ . '../view/partial/paginator.phtml',
];
```

(I have a couple more templates in mine due to having done some extra tutorials in my skeleton checkout.)

The only bit wrong with it is the missing `/` at the start of the quoted path, which I'll add now; otherwise, I'm getting all entries (and not missing the `application/index/index` as you observe).